messybrainz/fetch_artist_mbids.py

```diff
@@ -0,0 +1,117 @@
+import json
+import logging
+import re
+
+import messybrainz.default_config as config
+try:
+    import messybrainz.custom_config as config
+except ImportError:
+    pass
+
+from brainzutils import cache
+from brainzutils import musicbrainz_db
+from brainzutils.musicbrainz_db.exceptions import NoDataFoundException
+from messybrainz import db
+from sqlalchemy import text
+
+
+def check_valid_uuid(s):
+    """ Checks if the recording MBID is in the valid format or not.
+        Returns True if its a valid UUID else returns False.
+    """
+    if re.match(r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$', s):
+        return True
+    else:
+        return False
+
+
+def add_artist_mbids(recording_mbid, artist_mbids):
+    """ Inserts the artist_mbids corresponding to the recording_mbids
+        into the recording_artist table.
+    """
+    with db.engine.begin() as connection:
+        for artist_mbid in artist_mbids:
+            query = text("""INSERT INTO recording_artist (recording_mbid, artist_mbid)
+                            VALUES (:recording_mbid, :artist_mbid)
+                            RETURNING recording_mbid""")
+            result = connection.execute(query, {
+            "recording_mbid": recording_mbid,
+            "artist_mbid": artist_mbid,
+            })
+
+    return is_recording_mbid_present(recording_mbid)
+
+
+def is_recording_mbid_present(recording_mbid):
+    """
+        Check if recording MBID is already present in recording_artist table.
+        Returns True if recording MBID is present else False is returned.
+    """
+    with db.engine.begin() as connection:
+        query = text("""SELECT recording_mbid FROM recording_artist
+                        WHERE recording_mbid = :recording_mbid""")
+        result = connection.execute(query, {
+        "recording_mbid": recording_mbid,
+        })
+
+        if result.rowcount:
+            return True
+
+    return False
+
+
+def fetch_artist_mbids(recording_mbid):
+    """ Fetches artist MBIDs from the MusicBrainz database for the recording MBID.
+        And inserts the artist MBIDs into the recording_artist table.
+    """
+    try :
+        recording = mb_recording.get_recording_by_mbid(recording_mbid, include=['artists'])
+    except NoDataFoundException:
+        return False
+
+    artist_mbids = []
+    for artist in recording['artists']:
+        artist_mbids.append(artist['id'])
+
+    result = add_artist_mbids(recording_mbid, artist_mbids)
+
+    return result
+
+
+def fetch_artist_mbids_for_all_recording_mbids():
+    """ Fetches artist MBIDs from the musicbrainz database for the recording MBIDs
+        in the recording_json table submitted while submitting a listen.
+        Returns the number of recording MBIDs that were processed and number of
+        recording MBIDs that were added to the recording_artist table.
+    """
+    # Init databases
+    db.init_db_engine(config.SQLALCHEMY_DATABASE_URI)
```

Comment:
Why are the init steps being done in a function to fetch artist mbids? It should be done by the main function that sets up the correct environment...